File: roman2integer/Solution.java
class Solution {
    public int romanToInt(String s) {
        Map map 
            = Map.of("I",Integer.valueOf(1), 
                     "V",Integer.valueOf(5), 
                     "X",Integer.valueOf(10),
                     "L",Integer.valueOf(50),
                     "C",Integer.valueOf(100),
                     "D",Integer.valueOf(500),
                     "M",Integer.valueOf(1000));

        char[] ch = s.toCharArray();
        int result = 0;

        for(int i = 0; i < ch.length; i++){
            
            if(i == ch.length -1){
                result += (int) map.get(String.valueOf(ch[i]));
            } else if((int) map.get(String.valueOf(ch[i])) >= (int) map.get(String.valueOf(ch[i+1]))){
                result += (int) map.get(String.valueOf(ch[i]));
            } else {
                result += (int) map.get(String.valueOf(ch[i+1])) - (int) map.get(String.valueOf(ch[i]));
                i++; 
            }
        }

        return result; 
    }
}

File: roman2integer/Solution.php
<?php
class Solution {

    /**
     * @param String $s
     * @return Integer
     */
    function romanToInt($s) {
        $array = array(
            "I" => 1,
            "V" => 5,
            "X" => 10,
            "L" => 50,
            "C" => 100,
            "D" => 500,
            "M" => 1000
             );

        $char = str_split($s);
        $result = 0;

        for($i = 0; $i < count($char); $i++){
            
            if($i == count($char) -1){
                $result += $array[$char[$i]];
            } else if($array[$char[$i]] >= $array[$char[$i+1]]){
                $result += $array[$char[$i]];
            } else {
                $result += $array[$char[$i+1]] - $array[$char[$i]];
                $i++; 
            }
        }

        return $result;
    }
}